packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -83,6 +85,11 @@ export class BlobManager {
             async () => this.getStorage().readBlob(response.id),
         );
 
+        if (this.runtime.attachState !== AttachState.Attached) {
+            this.blobIds.add(response.id);
```

Comment:
There is no code that would do anything (correctly) with these blobs on file creation.
I assume this is point in time.
I'd add assert / fail attachment process if one tries to attach the file with such blobs, as currently there is no way to accomplish that correctly. That way developers may experiment (and ship) detached flow just fine as long as they never try to complete attachment process

Reply:
> I assume this is point in time.

That's correct.  We're delivering the feature in two phase.  The first supporting detached flow only, with support for attaching as a fast fallow up.

(The assert is a good idea, since we may end up deploying with phase 1 for a short window.)

Reply:
@wes-carlson, please add assert this API is not called in attaching state.